ethereum/pow/api/src/main/java/tech/pegasys/teku/ethereum/pow/api/DepositTreeSnapshot.java

```diff
@@ -19,62 +19,43 @@
 import org.apache.tuweni.bytes.Bytes32;
 import tech.pegasys.teku.infrastructure.json.types.DeserializableTypeDefinition;
 import tech.pegasys.teku.infrastructure.ssz.SszList;
-import tech.pegasys.teku.infrastructure.ssz.containers.Container3;
-import tech.pegasys.teku.infrastructure.ssz.containers.ContainerSchema3;
+import tech.pegasys.teku.infrastructure.ssz.containers.Container5;
 import tech.pegasys.teku.infrastructure.ssz.primitive.SszBytes32;
 import tech.pegasys.teku.infrastructure.ssz.primitive.SszUInt64;
-import tech.pegasys.teku.infrastructure.ssz.schema.SszListSchema;
-import tech.pegasys.teku.infrastructure.ssz.schema.SszPrimitiveSchemas;
 import tech.pegasys.teku.infrastructure.ssz.tree.TreeNode;
 import tech.pegasys.teku.infrastructure.unsigned.UInt64;
 
 public class DepositTreeSnapshot
-    extends Container3<DepositTreeSnapshot, SszList<SszBytes32>, SszUInt64, SszBytes32> {
+    extends Container5<
+        DepositTreeSnapshot, SszList<SszBytes32>, SszBytes32, SszUInt64, SszBytes32, SszUInt64> {
 
-  private static class DepositTreeSnapshotSchema
-      extends ContainerSchema3<DepositTreeSnapshot, SszList<SszBytes32>, SszUInt64, SszBytes32> {
-
-    protected DepositTreeSnapshotSchema() {
-      super(
-          "DepositTreeSnapshot",
-          namedSchema(
-              "finalized",
-              SszListSchema.create(SszPrimitiveSchemas.BYTES32_SCHEMA, Integer.MAX_VALUE)),
-          namedSchema("deposits", SszPrimitiveSchemas.UINT64_SCHEMA),
-          namedSchema("execution_block_hash", SszPrimitiveSchemas.BYTES32_SCHEMA));
-    }
-
-    @Override
-    public DepositTreeSnapshot createFromBackingNode(final TreeNode node) {
-      return new DepositTreeSnapshot(this, node);
-    }
-
-    @SuppressWarnings("unchecked")
-    public SszListSchema<SszBytes32, ? extends SszList<SszBytes32>> getFinalizedSchema() {
-      return (SszListSchema<SszBytes32, ? extends SszList<SszBytes32>>) getChildSchema(0);
-    }
-  }
-
-  private static final DepositTreeSnapshotSchema SCHEMA = new DepositTreeSnapshotSchema();
-
-  public static DeserializableTypeDefinition<DepositTreeSnapshot> getJsonTypeDefinition() {
-    return SCHEMA.getJsonTypeDefinition();
+  public static DeserializableTypeDefinition<DepositTreeSnapshot> getJsonTypeDefinition(
+      final DepositTreeSnapshotSchema schema) {
+    return schema.getJsonTypeDefinition();
   }
 
-  public static DepositTreeSnapshot fromBytes(final Bytes bytes) {
-    return SCHEMA.sszDeserialize(bytes);
+  public static DepositTreeSnapshot fromBytes(
+      final Bytes bytes, final DepositTreeSnapshotSchema schema) {
+    return schema.sszDeserialize(bytes);
   }
 
   public DepositTreeSnapshot(
-      final List<Bytes32> finalized, final long deposits, final Bytes32 executionBlockHash) {
+      final DepositTreeSnapshotSchema schema,
```

Comment:
We shouldn't have public constructors that take the schema as a param - this class should just know what it's schema is.